tests/test_checker.py

```diff
@@ -0,0 +1,118 @@
+
+    async def test_requires_invite_missing(self) -> None:
+        """Tests that the 3PID can register if the endpoint responds without a
+        "requires_invite" parameter (since it defaults to false).
+        """
+        await self._test_check_if_allowed(Addresses.REQUIRES_INVITE_MISSING, True)
+
+    async def test_no_invite_needed(self) -> None:
+        """Tests that the 3PID can register if no invite is needed."""
+        await self._test_check_if_allowed(Addresses.NO_INVITE_NEEDED, True)
+
+    async def test_invited_missing(self) -> None:
+        """Tests that the 3PID cannot register if the endpoint responds without an
+        "invited" parameter (since it defaults to false) and "requires_invite" is true.
+        """
+        await self._test_check_if_allowed(Addresses.INVITED_MISSING, False)
+
+    async def test_not_invited(self) -> None:
+        """Tests that the 3PID cannot register if the endpoint responds without a pending
+        invite and "requires_invite" is true.
+        """
+        await self._test_check_if_allowed(Addresses.NOT_INVITED, False)
+
+    async def test_invited(self) -> None:
+        """Tests that the 3PID can register if the endpoint responds with a pending invite
+        and "requires_invite" is true.
+        """
+        await self._test_check_if_allowed(Addresses.INVITED, True)
+
+    async def test_not_invited_registration(self) -> None:
+        """Tests that a 3PID that shouldn't be allowed to be associated with a local
+        account is only correctly blocked during user registration if the
+        "only_check_at_registration" configuration flag is set to true.
+        """
+        # While registering.
+        await self._test_check_if_allowed(
+            address=Addresses.NOT_INVITED,
+            expected_result=False,
+            only_registration=True,
+            registration=True,
+        )
+        # Not in a user registration.
+        await self._test_check_if_allowed(
+            address=Addresses.NOT_INVITED,
+            expected_result=True,
+            only_registration=True,
+            registration=False,
+        )
+
+    async def _test_check_if_allowed(
+        self,
+        address: str,
+        expected_result: bool,
+        only_registration: bool = False,
+        registration: bool = False,
+    ) -> None:
+        """Calls the "check_if_allowed" callback on a new instance of the module and
+        compare its return value with the expected result.
+        """
+        config = {"only_check_at_registration": only_registration}
+        module = create_module(config)
+        res = await module.check_if_allowed("email", address, registration)
+        self.assertEqual(res, expected_result)
+
+    async def test_config_missing_url(self) -> None:
+        """Tests that the module raises an error if configured without a URL."""
+        with self.assertRaises(ConfigError):
+            create_module({"url": None})
```

Comment:
Maybe also wort testing if no `url` key is given?

Reply:
Ah, yeah, this is what this was meant to test.